pulp_ansible/app/tasks/collections.py

```diff
@@ -272,32 +248,39 @@ def _get_url(page):
 
             return get_page_url(remote.url, page)
 
-        with ProgressBar(message="Parsing Pages from Galaxy Collections API") as progress_bar:
+        with ProgressBar(message="Parsing Galaxy Collections API") as progress_bar:
             url = _get_url(page_count)
             downloader = remote.get_downloader(url=url)
-            metadata = parse_metadata(await downloader.run())
-            metadata = filter_namespace(metadata, url)
+            initial_data = parse_metadata(await downloader.run())
 
-            count = len(self.collection_info) or metadata.get("count", 1)
+            count = len(self.collection_info) or initial_data.get("count", 1)
             page_count = math.ceil(float(count) / float(PAGE_SIZE))
-            progress_bar.total = page_count
+            progress_bar.total = count
             progress_bar.save()
 
-            yield metadata
-            progress_bar.increment()
-
             # Concurrent downloads are limited by aiohttp...
             not_done = set(
-                remote.get_downloader(url=_get_url(page)).run() for page in range(2, page_count + 1)
+                remote.get_downloader(url=_get_url(page)).run() for page in range(1, page_count + 1)
```

Comment:
This line could be broken up for readability.

Reply:
Black requires to put on the same line:
```bash
--- pulp_ansible/app/tasks/collections.py       2019-08-16 15:21:52.437710 +0000
+++ pulp_ansible/app/tasks/collections.py       2019-08-16 15:22:02.078186 +0000
@@ -258,12 +258,11 @@
             progress_bar.total = count
             progress_bar.save()
 
             # Concurrent downloads are limited by aiohttp...
             not_done = set(
-                remote.get_downloader(url=_get_url(page)).run()
-                for page in range(1, page_count + 1)
+                remote.get_downloader(url=_get_url(page)).run() for page in range(1, page_count + 1)
             )
 
             while not_done:
                 done, not_done = await asyncio.wait(not_done, return_when=asyncio.FIRST_COMPLETED)
                 for item in done:
would reformat pulp_ansible/app/tasks/collections.py
All done! 💥 💔 💥
```

Reply:
I agree I was thinking of something like:

```
not_done = set()
for page in range(1, page_count + 1):
    downloader = remote.get_downloader(url=_get_url(page))
    not_done.add(downloader.run())
```

Reply:
this is great, thanks @bmbouter